src/app/static/src/tests/components/projects/shareProject.test.ts

```diff
@@ -0,0 +1,159 @@
+        const wrapper = mount(ShareProject, {
+            propsData: {
+                project: {id: 1, name: "p1"}
+            },
+            store: createStore()
+        });
+
+        const link = wrapper.find("a");
+        link.trigger("click");
+        wrapper.find(Modal).find("input").setValue("bademail");
+        wrapper.find(Modal).findAll("button").at(0).trigger("click");
+        expect(wrapper.find(Modal).props("open")).toBe(true);
+        expect(wrapper.find(Modal).findAll(LoadingSpinner).length).toBe(0);
+        expect(wrapper.find(Modal).find("input").classes()).toContain("is-invalid");
+    });
+
+    it("if emails are valid, project is shared", (done) => {
+        const wrapper = mount(ShareProject, {
+            propsData: {
+                project: {id: 1, name: "p1"}
+            },
+            store: createStore()
+        });
+
+        const link = wrapper.find("a");
+        link.trigger("click");
+        wrapper.find(Modal).find("input").setValue("goodemail@gmail.com");
+        wrapper.find(Modal).findAll("button").at(0).trigger("click");
+        expect(wrapper.find(Modal).props("open")).toBe(true);
+        expect(wrapper.find(Modal).findAll(LoadingSpinner).length).toBe(1);
+
+        setTimeout(() => {
+            expect(wrapper.find(Modal).props("open")).toBe(false);
+            expect(wrapper.find(Modal).findAll(LoadingSpinner).length).toBe(0);
+            done();
+        }, 200)
+    });
+
+    it("translates header", () => {
+        const store = createStore();
+        const wrapper = shallowMount(ShareProject, {
+            propsData: {
+                project: {id: 1, name: "p1"}
+            },
+            store: store
+        });
+
+        const link = wrapper.find("a");
+        link.trigger("click");
+        expectTranslated(wrapper.find(Modal).find("h4"), "Share project", "Partagez projet", store);
+    });
+
+    it("translates instructions", () => {
+        const store = createStore();
+        const wrapper = shallowMount(ShareProject, {
+            propsData: {
+                project: {id: 1, name: "p1"}
+            },
+            store: store
+        });
+
+        const link = wrapper.find("a");
+        link.trigger("click");
+        const expectedEnglish  = "This will create a copy of p1 for the given users." +
+        "Please enter the email address or comma separated list of email addresses you would like to share" +
+        "this project with. These email addresses must be already registered with Naomi."
+
+        const expectedFrench = "Cela créera une copie de p1 pour les utilisateurs désignés." +
+            "Veuillez entrer une adresse e-mail ou une liste d'adresses séparées par des virgules " +
+            "avec lesquelles vous souhaitez partager ce projet. Ces addresses e-mails doivent être déjà enregistrées dans Naomi."
+
+        expectTranslated(wrapper.find(Modal).find("#instructions"), expectedEnglish, expectedFrench, store);
```

Comment:
Ah nice one, I did add a ticket for putting his in everywhere. https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-1789

Reply:
Great, thanks :) Once this is merged we can re-use this utility elsewhere